pennylane/transforms/core/transform_program.py

```diff
@@ -155,6 +158,8 @@ def push_back(self, transform_container: TransformContainer):
             raise TransformError("Only transform container can be added to the transform program.")
 
         # Program can only contain one informative transform and at the end of the program
+        if self.is_informative:
+            raise TransformError("The transform program already has an informative transform.")
```

Comment:
I don't think this is necessary. Checking if a program has a final transform also covers informative transforms.